File: 462-minimum-moves-to-equal-array-elements-ii/462-minimum-moves-to-equal-array-elements-ii.py
class Solution:
    def minMoves2(self, nums: List[int]) -> int:
        # # dumb algorithm
        # median = sorted(nums)[len(nums) // 2]
        # return sum(abs(num - median) for num in nums)
    
        # using bitwise inverse
        nums.sort()
        return sum(nums[~i] - nums[i] for i in range(len(nums) // 2))

File: 462-minimum-moves-to-equal-array-elements-ii/462-minimum-moves-to-equal-array-elements-ii.cpp
class Solution {
public:
    int minMoves2(vector<int>& nums) {
        int m = nums.size() >> 1;
        nth_element(nums.begin(), nums.begin() + m, nums.end());
        int ans = 0, mid = nums[m];
        for (int& x: nums) ans += abs(x - mid);
        return ans;
    }
};